org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/WRecordIDDialog.java

```diff
@@ -189,31 +205,40 @@ public void onEvent(Event event) throws Exception {
 			if(event.getTarget().equals(okBtn)) {
 				// set the selected values to the editors
 				editor.setAD_Table_ID(tableIDEditor.getValue());
-				editor.setValue(recordsEditor.getValue());
+				if(recordsEditor != null)
+					editor.setValue(recordsEditor.getValue());
 				onClose();
 			}
 			else if(event.getTarget().equals(cancelBtn)) {
 				// dismiss the selected values
 				onClose();
 			}
 		}
-		if(event.getName().equalsIgnoreCase(Events.ON_SELECT)) {
-			if (event.getTarget().equals(tableIDEditor.getComponent())) {
-				// the Record_ID should be cleared when a different AD_Table_ID is selected
-				if (recordsEditor != null) {
-					recordsEditor.setValue(null);
-					recordsEditorLabel.detach();
-					recordsEditor.getComponent().detach();
-				}
-				int tableID = Integer.parseInt(Objects.toString(tableIDEditor.getValue(), "-1"));
-				if(tableID > 0) {
-			    	recordsEditor = new WSearchEditor("Record_ID", false, false, true, editor.getRecordsLookup(tableID));
+	}
+	
+	@Override
+	public void valueChange(ValueChangeEvent evt) {
+		if (evt.getSource().equals(tableIDEditor)) {
+			// the Record_ID should be cleared when a different AD_Table_ID is selected
+			if (recordsEditor != null) {
+				recordsEditor.setValue(null);
+				recordsEditorLabel.detach();
+				recordsEditor.getComponent().detach();
+			}
+			int tableID = Integer.parseInt(Objects.toString(evt.getNewValue(), "-1"));
+			if(tableID > 0) {
+				MLookup recordsLookup = editor.getRecordsLookup(tableID);
+				if(recordsLookup != null) {
+					recordsEditor = tableID > 0 ? new WSearchEditor("Record_ID", false, false, true, recordsLookup) : null;
```

Comment:
again, duplicate checking for tableID > 0

Reply:
multiple tableID > 0 checks fixed